pkg/asset/installconfig/platformprovisioncheck.go

```diff
@@ -142,7 +142,11 @@ func (a *PlatformProvisionCheck) Generate(ctx context.Context, dependencies asse
 			return err
 		}
 
-		err = powervsconfig.ValidatePreExistingDNS(client, ic.Config, ic.PowerVS)
+		metadata, err := powervsconfig.NewMetadata(ic.Config)
+		if err != nil {
+			return err
+		}
```

Comment:
I don't think you need this, metadata has already been initialized as per comment above.

Reply:
I clearly remember this is what I had to add to fix the last segfault I actually encountered with ALL the other proposed changes in place.

Reply:
After offline discussion, we decided to go with just-in-time initialization (but cleaner), than relying on embedding `NetMetadata()` at places that are needed like here. As a result, this file is no longer part of this PR.